server/resp/src/main/java/org/infinispan/server/resp/serialization/lua/LuaResponseWriter.java

```diff
@@ -0,0 +1,221 @@
+          */
+         lua.checkStack(3);
+         lua.newTable();
+         lua.push("set");
+         lua.newTable();
+         for (Object o : set) {
+            contentType.serialize(o, this);
+            lua.checkStack(1);
+            lua.push(true);
+            lua.setTable(-3);
+         }
+         lua.setTable(-3);
+         lua.push("len");
+         lua.push(set.size());
+         lua.setTable(-3);
+      }
+   }
+
+   @Override
+   public <T> void set(Set<?> set, JavaObjectSerializer<T> serializer) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
+   }
+
+   @Override
+   public void map(Map<?, ?> map) {
+      map(map, Resp3Type.BULK_STRING); // This is currently only called with an empty map, so we don't care about the type
+   }
+
+   @Override
+   public void map(Map<?, ?> map, Resp3Type contentType) {
+      map(map, contentType, contentType);
+   }
+
+   @Override
+   public void map(Map<?, ?> map, Resp3Type keyType, Resp3Type valueType) {
+      if (map == null) {
+         nulls();
+      } else {
+         /*
+          * A map is represented as a lua table with two elements: 'map', containing a table
+          * with the map entries, and `len` containing the size of the set.
+          */
+         lua.checkStack(3);
+         lua.newTable();
+         lua.push("map");
+         lua.newTable();
+         for (Map.Entry<?, ?> entry : map.entrySet()) {
+            keyType.serialize(entry.getKey(), this);
+            valueType.serialize(entry.getValue(), this);
+            lua.setTable(-3);
+         }
+         lua.setTable(-3);
+         lua.push("len");
+         lua.push(map.size());
+         lua.setTable(-3);
+      }
+   }
+
+   @Override
+   public void map(Map<?, ?> map, SerializationHint.KeyValueHint keyValueHint) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
+   }
+
+   @Override
+   public void error(CharSequence value) {
+      lua.checkStack(3);
+      luaPushError(lua, value.toString());
+      lua.push("ignore_error_stats_update");
+      lua.push(true);
+      lua.setTable(-3);
+   }
+
+   @Override
+   public void error(Throwable t) {
+      error(filterCause(t).getMessage());
+   }
+
+   @Override
+   public <T> void write(T object, JavaObjectSerializer<T> serializer) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
+   }
+
+   @Override
+   public <T> void write(JavaObjectSerializer<T> serializer) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
+   }
+
+   @Override
+   public void serialize(Object object) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
+   }
+
+   @Override
+   public void writeNumericPrefix(byte symbol, long number, int additionalWidth) {
+      throw new UnsupportedOperationException(CallerId.getCallerMethodName(1));
```

Comment:
Should it be no-op? I see the method is called directly by some commands in the sorted set.

Reply:
Agreed, I don't feel comfortable with any of these methods throwing an exception tbh.

Reply:
I've rewritten most of the commands doing this to use proper RESP structures, instead of doing things by hand. The only significant leftover is `EXEC`. It looks like I should rewrite it to collect all responses into an array and write it at the end